en/services/parameter_ext.md

```diff
@@ -0,0 +1,238 @@
+<span id="PARAM_EXT_VALUE"></span>[PARAM_EXT_VALUE](../messages/common.md#PARAM_EXT_VALUE) | Emit the value of a parameter, following a [PARAM_EXT_REQUEST_LIST](#PARAM_EXT_REQUEST_LIST) or [PARAM_EXT_REQUEST_READ](#PARAM_EXT_REQUEST_READ). The message includes `param_count` and `param_index` which the recipient can use to track received parameters and re-request missing parameters after a timeout.
+<span id="PARAM_EXT_REQUEST_READ"></span>[PARAM_EXT_REQUEST_READ](../messages/common.md#PARAM_EXT_REQUEST_READ) | Request the value of a specific parameter using either its `param_id` or `param_index`. Expects a response in a [PARAM_EXT_VALUE](#PARAM_EXT_VALUE).
+<span id="PARAM_EXT_SET"></span>[PARAM_EXT_SET](../messages/common.md#PARAM_EXT_SET) | Set a parameter value. Expects immediate response [PARAM_EXT_ACK](#PARAM_EXT_ACK) with result indicating success, failure, or that the request is still in progress (`PARAM_ACK_IN_PROGRESS`). If in progress, additional update [PARAM_EXT_ACK](#PARAM_EXT_ACK) messages are expected.
+<span id="PARAM_EXT_ACK"></span>[PARAM_EXT_ACK](../messages/common.md#PARAM_EXT_ACK) | Response from a [PARAM_EXT_SET](#PARAM_EXT_SET) message, which indicates whether the value was accepted (set), failed, setting is still in progress, or that the specified parameter is invalid/unsupported.
+
+Enum | Description
+-- | --
+<span id="MAV_PARAM_EXT_TYPE"></span>[MAV_PARAM_EXT_TYPE](../messages/common.md#MAV_PARAM_EXT_TYPE) | Specifies the datatype of a MAVLink extended parameter (parameter values are [encoded](parameter_encoding) within the a `char[128]` array in the messages). This type conveys the *real type* of the encoded parameter value, e.g. `MAV_PARAM_EXT_TYPE_REAL32`.
+<span id="PARAM_ACK"></span>[PARAM_ACK](../messages/common.md#PARAM_ACK) | Request acknowledgment status value, sent in an [PARAM_EXT_ACK](#PARAM_EXT_ACK) as a response to a [PARAM_EXT_SET](#PARAM_EXT_SET) message. A request can be accepted, fail, in-progress, or unsupported (indicating the specified parameter does not exist or has an invalid value or value type).
+
+
+## Parameter Encoding {#parameter_encoding}
+
+Parameters names/ids are set in the `param_id` field of messages where they are used.
+The `param_id` string can store up to 16 characters. 
+The string is terminated with a NULL (`\0`) character if there are less than 16 human-readable chars, and without a null termination byte if the length is exactly 16 chars.
+
+> **Note** Names (as above) are the same as for the [Parameter Protocol](../services/parameter.md#parameter_encoding).
+  Values (below) are encoded differently.
+
+Values are byte-wise encoded *within* the `param_value` field, which is a `char[128]`. <!-- how are they encoded? -->
```

Comment:
@dogmaphobic 
- How are values encoded in the char[128] (if you see the note below, it used to be a byte-wise conversion into a float for numeric values). 
- Is it just a byte-wise copy into the appropriate sized number of bytes in the message? Any example code to point to? Any "helper functions"?
- What about for strings - how are they represented?